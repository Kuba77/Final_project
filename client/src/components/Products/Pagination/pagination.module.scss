@import "../../../styles/config/variables.scss";
.pagination_buttons {
  width: 300px;
  margin: 0 auto;
  margin-top: 100px;
  &__container {
    display: flex;
    justify-content: space-evenly;
  }
  &__item a {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 50px;
    height: 50px;
    background-color: rgba(141, 40, 173, 0.15);
    border-radius: 5px;
    font-size: $small-font-size;
    color: $first-color;
  }
  &__item a:hover {
    background-color: $first-color;
    cursor: pointer;
    color: $body-color;
  }
}
