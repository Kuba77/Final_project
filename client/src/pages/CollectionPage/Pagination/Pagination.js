import React from "react";
import classes from "./pagination.module.scss";
const Pagination = ({ productsInPage, totalProducts, paginate}) => {
  const pageNumbers = [];

  for (let i = 1; i <= Math.ceil(totalProducts / productsInPage); i++) {
    pageNumbers.push(i);
  }
  return (
    <div className={classes.pagination_buttons}>
      <ul className={classes.pagination_buttons__container}>
        {pageNumbers.map((number) => (
          <li key={number} className={classes.pagination_buttons__item}>
            <a href="#" onClick={() => paginate(number)}>
              {number}
            </a>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Pagination;
