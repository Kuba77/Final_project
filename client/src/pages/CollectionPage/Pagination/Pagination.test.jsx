import React from "react";
import Pagination from "./Pagination";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

const productsInPage = 10;
const totalProducts = 30;
const onClickHandlerPagination = jest.fn();

describe("Testing of pagination component", () => {
  it("rendering pagination component with props", () => {
    render(
      <Pagination
        productsInPage={productsInPage}
        totalProducts={totalProducts}
      />
    );
    screen.debug();
  });
  it("rendering pagination component without props", () => {
    render(<Pagination />);
    screen.debug();
  });
  it("pagination renders like list", () => {
    render(<Pagination />);
    expect(screen.getByRole("list")).toBeInTheDocument();
  });
  it("renders with snapshots", () => {
    const listSnap = render(
      <Pagination
        productsInPage={productsInPage}
        totalProducts={totalProducts}
      />
    );
    expect(listSnap).toMatchSnapshot();
  });
  it("renders with snapshots and empty props", () => {
    const listSnap = render(<Pagination />);
    expect(listSnap).toMatchSnapshot();
  });
  it("onclick cheking", () => {
    render(
      <Pagination
        paginate={onClickHandlerPagination}
        productsInPage={productsInPage}
        totalProducts={totalProducts}
      />
    );
    const btnPaginate = screen.getByText(1);
    fireEvent.click(btnPaginate);
  });
});
