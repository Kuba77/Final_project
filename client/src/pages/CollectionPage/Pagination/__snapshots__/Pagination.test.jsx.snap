// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of pagination component renders with snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pagination_buttons"
      >
        <ul
          class="pagination_buttons__container"
        >
          <li
            class="pagination_buttons__item"
          >
            <a
              data-testid="1"
              id="1"
            >
              1
            </a>
          </li>
          <li
            class="pagination_buttons__item"
          >
            <a
              data-testid="2"
              id="2"
            >
              2
            </a>
          </li>
          <li
            class="pagination_buttons__item"
          >
            <a
              data-testid="3"
              id="3"
            >
              3
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pagination_buttons"
    >
      <ul
        class="pagination_buttons__container"
      >
        <li
          class="pagination_buttons__item"
        >
          <a
            data-testid="1"
            id="1"
          >
            1
          </a>
        </li>
        <li
          class="pagination_buttons__item"
        >
          <a
            data-testid="2"
            id="2"
          >
            2
          </a>
        </li>
        <li
          class="pagination_buttons__item"
        >
          <a
            data-testid="3"
            id="3"
          >
            3
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing of pagination component renders with snapshots and empty props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pagination_buttons"
      >
        <ul
          class="pagination_buttons__container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pagination_buttons"
    >
      <ul
        class="pagination_buttons__container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
